src/MessagePack.UnityClient/Assets/Scripts/MessagePack/Internal/ExpressionUtility.cs

```diff
@@ -11,10 +11,14 @@ internal static class ExpressionUtility
     {
         private static MethodInfo GetMethodInfoCore(LambdaExpression expression)
         {
+#if NET6_0_OR_GREATER
+            ArgumentNullException.ThrowIfNull(expression, nameof(expression));
```

Comment:
I don't want a bunch of `#if` statements for something so trivially addressible in another way. Just create an `internal static` helper class that has the if and throw check inside it, and use that in all the places you've changed here so we can do without the `#if`/`#else` blocks.